packages/espressocash_app/lib/features/kyc_sharing/data/kyc_repository.dart

```diff
@@ -0,0 +1,143 @@
+import 'dart:typed_data';
+
+import 'package:flutter/material.dart';
+import 'package:injectable/injectable.dart' hide Order;
+import 'package:kyc_client_dart/kyc_client_dart.dart';
+
+import '../../accounts/auth_scope.dart';
+import '../../accounts/models/ec_wallet.dart';
+
+@Singleton(scope: authScope)
+class KycRepository extends ChangeNotifier {
+  KycRepository(this._ecWallet);
+
+  final ECWallet _ecWallet;
+
+  late final KycUserClient _kycUserClient;
+
+  late String _authPublicKey = '';
+  late String _rawSecretKey = '';
+
+  @PostConstruct()
+  Future<void> init() async {
+    _kycUserClient = KycUserClient(
+      sign: (data) async {
+        final signature =
+            await _ecWallet.sign([Uint8List.fromList(data.toList())]);
+
+        return signature.first;
+      },
+    );
+
+    await _kycUserClient.init(walletAddress: _ecWallet.publicKey.toString());
```

Comment:
This will initialize in background, and if it fails, client will stay in the broken state.

Reply:
Should we do some automatic retry in that case?

Reply:
Yeah, I would probably expect the whole initialization to be cached future, that resolves once it's initialized and tries periodically otherwise.